docs/scopes.md

```diff
@@ -0,0 +1,199 @@
+# OAuth Scopes
+
+Each authorization grant in OAuth has an associated "scope",
+a list containing one or more "scope values"
+that indicate what capabilities the granted token will have.
+Each individual scope value indicates a particular capability,
+such as the ability to read or write profile data,
+or to access the user's data in a particular service.
+
+As defined in [RFC6749 Section 3.3](https://tools.ietf.org/html/rfc6749#section-3.3),
+the scope of a token is expressed as
+a list of space-delimited, case-sensitive strings,
+and it is left up to the service
+to define the format and semantics
+of the individual scope values
+that make up this string.
+
+This document defines the scope values
+accepted in the Firefox Accounts ecosystem,
+and the rules for parsing and validating them.
+
+## Short-name scope values
+
+FxA supports a small set of "short-name" scope values
+that are identified by a short English word.
+These correspond either to
+scope values defined by external specifications
+(e.g. OpenID Connect),
+or to legacy scope values
+introduced during early development.
+
+**No new short-name scope values should be added.**
+Instead we prefer to use URLs for new scope values,
+both to ensure uniqueness
+and to simplify parsing rules.
+
+Short-name scope values imply read-only access by default,
+with write access indicated by the suffix ":write".
+The may also have "sub-scopes"
+to indicate finer-grained access control.
+Each name component may contain only ascii alphanumeric characters
+and the underscore.
+
+For example:
+
+* `profile` indicates read-only access
+  to the user's profile data.
+* `profile:write` indicates read/write access
+  to the user's profile data.
+* `profile:display_name` indicates read-only access
+  to the user's display name, but not any other
+  profile data.
+* `profile:email:write` indicates read/write access
+  to the user's email address.
+
+The following short-name scope values are recognized
+in the FxA ecosystem.
+
+### Profile data
+
+* `profile`: access the user's profile data.
+* `profile:uid`: access the user's opaque user id.
+* `profile:email`: access the user's email address.
+* `profile:locale`: access the user's locale.
+* `profile:avatar`: access the user's avatar picture.
+* `profile:display_name`: access the user's human-readable display name.
+* `profile:amr`: access information about the user's authentication methods and 2FA status.
```

Comment:
Should there be any indication on whether any of these are read vs read/write? uid/locale/amr I imagine are read-only.

Reply:
It's true that the `amr` cannot be modified, but "profile:amr:write" is still a well-formed scope with well-defined semantics (that doesn't happen to grant you any addition powers over "profile:amr".